bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/handler/EcobeeAccountBridgeHandler.java

```diff
@@ -0,0 +1,285 @@
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.auth.client.oauth2.OAuthFactory;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.ecobee.internal.api.EcobeeApi;
+import org.openhab.binding.ecobee.internal.config.EcobeeAccountConfiguration;
+import org.openhab.binding.ecobee.internal.discovery.ThermostatDiscoveryService;
+import org.openhab.binding.ecobee.internal.dto.SelectionDTO;
+import org.openhab.binding.ecobee.internal.dto.thermostat.ThermostatDTO;
+import org.openhab.binding.ecobee.internal.dto.thermostat.ThermostatUpdateRequestDTO;
+import org.openhab.binding.ecobee.internal.dto.thermostat.summary.SummaryResponseDTO;
+import org.openhab.binding.ecobee.internal.function.FunctionRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EcobeeAccountBridgeHandler} is responsible for managing
+ * communication with the Ecobee API.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class EcobeeAccountBridgeHandler extends BaseBridgeHandler {
+
+    private static final int REFRESH_STARTUP_DELAY_SECONDS = 3;
+    private static final int REFRESH_INTERVAL_SECONDS = 1;
+    private static final int DISCOVERY_INTERVAL_SECONDS = 300;
+    private static final int DISCOVERY_INITIAL_DELAY_SECONDS = 10;
+    private static final int DEFAULT_REFRESH_INTERVAL_NORMAL_SECONDS = 20;
+    private static final int DEFAULT_REFRESH_INTERVAL_QUICK_SECONDS = 5;
+    private static final int DEFAULT_API_TIMEOUT_SECONDS = 20;
+
+    private final Logger logger = LoggerFactory.getLogger(EcobeeAccountBridgeHandler.class);
+
+    private final OAuthFactory oAuthFactory;
+    private final HttpClient httpClient;
+
+    private @NonNullByDefault({}) EcobeeApi api;
+    private @NonNullByDefault({}) String apiKey;
+    private int refreshIntervalNormal;
+    private int refreshIntervalQuick;
+    private int apiTimeout;
+    private boolean discoveryEnabled;
+    private int discoveryInterval;
+
+    private final Map<String, EcobeeThermostatBridgeHandler> thermostatHandlers = new ConcurrentHashMap<>();
+    private final Set<String> thermostatIds = new CopyOnWriteArraySet<>();
+
+    private @Nullable Future<?> refreshThermostatsJob;
+    private final AtomicInteger refreshThermostatsCounter = new AtomicInteger(REFRESH_STARTUP_DELAY_SECONDS);
```

Comment:
Could you explain how this is being used? It isn't entirely clear to me from reading the code...

Reply:
Rather than manage multiple scheduled jobs for normal polling, quick polling, and discovery, I have a single job that runs every second. I then use `refreshThermostatsCounter` and `discoveryCounter` to control the rate at which thermostat refresh and thermostat discovery will be performed.

It's especially useful when I need to flip between the normal refresh interval and the quick refresh interval. This way I'm not constantly starting and stopping scheduled jobs.

The overhead of running every second is very low, except when refresh and/or discovery needs to run.

Reply:
Thanks for explaining it.